src/azul/indexer/index_service.py

```diff
@@ -287,16 +287,24 @@ def _read_aggregates(self, entities: CataloguedTallies) -> Dict[CataloguedEntity
                 for coordinate in coordinates
             ]
         }
+        mandatory_source_fields = set()
+        for coordinate in coordinates:
+            aggregate_cls = self.metadata_plugin(coordinate.entity.catalog).aggregate_class()
```

Comment:
I think we should first collect all catalogs in a set and then iterate over that set to populate mandatory_source_fields. 

We may also want to extract `self.metadata_plugin(catalog).aggregate_class()` into a method and think about applying lru_cache to that method.